web/models/stable_diffusion/opt_params.py

```diff
@@ -25,156 +26,94 @@
-                if args.max_length == 64:
-                    model_name = "unet_19dec_v2p1base_fp16_64"
-                else:
-                    model_name = "unet2base_8dec_fp16"
-            if args.version == "v2.1":
-                model_name = "unet2_14dec_fp16"
+
+    try:
+        model_name = models_db[model_key]
+    except KeyError:
+        raise Exception(f"{model_key} is not present in the models database")
+
+    return model_name, iree_flags
+
+
+def get_unet():
+    # Tuned model is present only for `fp16` precision.
+    is_tuned = "/tuned" if args.use_tuned else "/untuned"
+    model_key = f"SD/{args.version}/unet/{args.precision}/length_{args.max_length}{is_tuned}"
+    model_name, iree_flags = get_params(model_key)
+    if args.use_tuned:
+        bucket = "gs://shark_tank/vivian"
+        return get_shark_model(bucket, model_name, iree_flags)
+    else:
+        bucket = "gs://shark_tank/stable_diffusion"
+        if args.precision == "fp16":
             iree_flags += [
                 "--iree-flow-enable-padding-linalg-ops",
                 "--iree-flow-linalg-ops-padding-size=32",
                 "--iree-flow-enable-conv-img2col-transform",
             ]
-            if args.import_mlir:
-                return get_unet_mlir(model_name, iree_flags)
-            return get_shark_model(bucket, model_name, iree_flags)
-
-    # Tuned model is not present for `fp32` case.
-    if args.precision == "fp32":
-        bucket = "gs://shark_tank/stable_diffusion"
-        model_name = "unet_1dec_fp32"
-        iree_flags += [
-            "--iree-flow-enable-conv-nchw-to-nhwc-transform",
-            "--iree-flow-enable-padding-linalg-ops",
-            "--iree-flow-linalg-ops-padding-size=16",
-        ]
+        elif args.precision == "fp32":
+            iree_flags += [
+                "--iree-flow-enable-conv-nchw-to-nhwc-transform",
+                "--iree-flow-enable-padding-linalg-ops",
+                "--iree-flow-linalg-ops-padding-size=16",
+            ]
         if args.import_mlir:
             return get_unet_mlir(model_name, iree_flags)
         return get_shark_model(bucket, model_name, iree_flags)
 
-    if args.precision == "int8":
-        bucket = "gs://shark_tank/prashant_nod"
-        model_name = "unet_int8"
+
+def get_vae():
+    # Tuned model is present only for `fp16` precision.
+    is_tuned = "/tuned" if args.use_tuned else "/untuned"
+    is_base = "/base" if args.use_base_vae else ""
+    model_key = (
+        f"SD/{args.version}/vae/{args.precision}/length_77{is_tuned}{is_base}"
+    )
+    model_name, iree_flags = get_params(model_key)
+    if args.use_tuned:
+        bucket = "gs://shark_tank/vivian"
         iree_flags += [
             "--iree-flow-enable-padding-linalg-ops",
             "--iree-flow-linalg-ops-padding-size=32",
+            "--iree-flow-enable-conv-img2col-transform",
+            "--iree-flow-enable-conv-winograd-transform",
```

Comment:
@yzhang93 can correct me if I'm wrong but I don't think we should be doing winograd like this.

Reply:
@qedawkins The current VAE tuned model is just with winograd attributes. So we'll need to use these flags.

Reply:
It should do winograd regardless of the flag if it's annotated though, so this is forcing any non-winograd convs to go down winograd (although since it's tuned there might not be any convs left, so this flag doesn't matter?)

Reply:
@qedawkins I'm not sure I understand. Currently this model is the original mlir + some convs marked `iree_winograd_conv`. The mlir wasn't dump after certain pass, and doesn't have lowering configs. So I think these flags are needed for lowering?

Reply:
@yzhang93 @qedawkins So should we remove these flags from tuned, as for tuned the optimizations should have been done already?

Reply:
Keeping the flags as it is.